planner/core/property_cols_prune.go

```diff
@@ -165,13 +165,27 @@ func (p *LogicalProjection) PreparePossibleProperties(schema *expression.Schema,
 	return childProperties
 }
 
+func clonePossibleProperties(props [][]*expression.Column) [][]*expression.Column {
+	res := make([][]*expression.Column, len(props))
+	for i, prop := range props {
+		clonedProp := make([]*expression.Column, len(prop))
+		for j, col := range prop {
+			clonedProp[j] = col.Clone().(*expression.Column)
```

Comment:
Do we need a deep copy here?

Reply:
I don't think so, I didn't see places where the referenced structs in `Column` are modified and impact others, at least for now.

Reply:
We should use `clonedProp[j] = col` then.

Reply:
Why? I mean we need a shallow copy instead of a deep copy on `Column`, not on the slice, and `Clone()` is a shallow copy for `Column`.